src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle'
 }
 
-export class Triangle {
-
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea: () => number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b < 0 || c <= 0) {
+      throw new Error('The lenght of slide can not be less than zero');
+    }
+
+    if (a + b <= c || b + c <= a || a + c <= c) {
+      throw new Error(`sides ${a}, ${b} and ${c}can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const semiperimeter = (1 / 2) * (this.a + this.b + this.c);
```

Comment:
I say just multiply with 0.5 here
```js
const semiperimeter = 0.5 * (this.a + this.b + this.c);
```

in case this code is being used for creating a million triangles :D